File: structy/01-max-array-and-string/04-most-frequent-char/02.js
const mostFrequentChar = (s) => {
    const count = {};
    
    for (let char of s) {
      if (!(char in count)) {
        count[char] = 0;
      }
      count[char] += 1
    }
    
    let best = null;
    for (let char of s) {
      if (best === null || count[char] > count[best]) {
        best = char;
      }
    }
    return best;
  };

console.log(mostFrequentChar('bookeeper')); // -> 'e'
console.log(mostFrequentChar('david')); // -> 'd'
console.log(mostFrequentChar('abby')); // -> 'b'
console.log(mostFrequentChar('mississippi')); // -> 'i'
console.log(mostFrequentChar('potato')); // -> 'o'
console.log(mostFrequentChar('eleventennine')); // -> 'e'
console.log(mostFrequentChar("riverbed")); // -> 'r'

File: structy/01-max-array-and-string/04-most-frequent-char/02.py


def most_frequent_char(s):
    count = {}
    for char in s:
        if char not in count:
            count[char] = 0
        count[char] += 1

    best = None

    for char in s: 
        if best is None or count[char] > count[best]:
            best = char 

    return best

print(most_frequent_char('bookeep')) # 'o'
print(most_frequent_char('bookeeper')) # 'e'
print(most_frequent_char('david')) # 'd'
print(most_frequent_char('abby')) # 'b'
print(most_frequent_char('mississippi')) # 'i'
print(most_frequent_char('potato')) # 'o'
print(most_frequent_char('eleventennine')) # 'e'
print(most_frequent_char("riverbed")) # 'r'